examples/python/create_dashboard_filter.py

```diff
@@ -0,0 +1,88 @@
+
+def main():
+    """This file creates a new dashboard filter, and applies that filtering to all tiles on the dashboard.
+    Dashboard elements listen on the same field that the dashboard filter is created from.
+    This example can be modified to create a filter on many dashboards at once if you've added a new field to your LookML,
+        dynamically generate dashboards, etc.
+    """
+    
+    # Update these with the relevant information for the filter you want to create, and the dashboard
+    dash_id = '<dashboard_id>'
+    filter_name = '<name_of_filter>'
+    filter_model = '<model_name>'
+    filter_explore = '<explore_name>'
+    filter_dimension = '<view_name.field_name>' # Requires a fully-scoped dimension
+
+    filter = create_filter(dash_id, filter_name, filter_model, filter_explore, filter_dimension)
+    elements = sdk.dashboard_dashboard_elements(dash_id)
+    for element in elements:
+        # Skip elements that are text tiles
+        # Add other restrictions here to skip updating individual tiles
+        if element.type != 'text':
+            update_elements_filters(element, filter)
+
+
+def create_filter(dash_id: str, filter_name: str, filter_model: str, filter_explore: str , filter_dimension: str ) -> DashboardFilter:
+    """Creates a dashboard filter object on the specified dashboard. Filters must be tied to a specific LookML Dimension.
+
+    Args:
+        dash_id (str): ID of the dashboard to create the filter on
+        name (str): Name/Title of the filter
+        model (str): Model of the dimension
+        explore (str): Explore of the dimension
+        dimension (str): Name of the dimension. Must be in the format 'view_name.field_name'
+    """
+
+    return sdk.create_dashboard_filter(
+        body=models.WriteCreateDashboardFilter(
+            dashboard_id=dash_id,
+            name=filter_name,
+            title=filter_name,
+            type='field_filter', # New dashboards are only compatible with field filters
+            model=filter_model,
+            explore=filter_explore,
+            dimension=filter_dimension,
+            # Add additional parameters as necessary for allowing multiple values, ui configuration, etc.
+            )
+        )
+
+def update_elements_filters(element: DashboardElement, filter: DashboardFilter) -> None:
+    """Updates a dashboard element's result maker to include a listener on the new dashboard filter.
+
+
+    Args:
+        element (DashboardElement): Dashboard element to update with the new filter
+        filter (DashboardFilter): Dashboard filter the element will listen to
+    """
+    # Keep track of the current result_maker and add to it, otherwise listeners for other filters would be removed
+    current_filterables = element.result_maker.filterables
+    for filterable in current_filterables:
+        new_listens = []
+        for each in filterable.listen:
+            new_listens.append(each)
+        # Add listener for new filter
```

Comment:
Is this where a customer might want to validate that the model/explore matches between the filter's field and the current query/filterable, before trying to add the listener? It seems like such a common thing to want to validate that maybe we should include that "if" statement as well, as a default piece of sample logic?

Reply:
Great idea. Originally I thought this may be more complicated due to the possibility of a merged result tile, but since we're already looping through the filterables in order to take care of merge results, it's an easy addition!